### PR TITLE
HPCC-18059 Dafilesrv ssl client to be certificateless

### PR DESCRIPTION
**Pull request by mckellyln**
This PR change is to make dafilesrv client SSL connections certificate-less, that is to not use any certificate info.

Signed-off-by: Mark Kelly <mark.kelly@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Full QA suite plus various QA queries with dafilesrv and the rest of hpcc at combinations of SSLNone/SSLFirst/UnsecureFirst/SSLOnly

### CODE REVIEW

**Comment by mckellyln** [comment]
@mayx, @afishbeck, @RussWhitehead, @jakesmith please review / comment.

**Comment by RussWhitehead** [comment]
@mckellyln Looks reasonable

**Comment by mayx** [comment]
@mckellyln looks good.

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -194,26 +194,30 @@ static class _securitySettings
...
+            secureContextClient.setown(createSecureSocketContext(type));
     }
+    int loglevel = SSLogNormal;
 #ifdef _DEBUG
-    return secureContext->createSecureSocket(sock, SSLogMax);
-#else
-    return secureContext->createSecureSocket(sock);
+    loglevel = SSLogMax;
 #endif
+    if (type == ServerSocket)
```
Comment (afishbeck) [inline common/remote/sockfile.cpp]: Trivial but you could have combined this if/else with the if/else above.

**Comment by mckellyln** [comment]
@jakesmith any comment ?

**Comment by jakesmith** [comment]
@mckellyln @afishbeck , the JIRA says "Client connections should either not use certificate info or not use it if it cannot read/open certificate and key file."

I think this change means that a client cannot use a certificate.
That is probably intentional, but at odds with the comment in the JIRA description which says 'or not use if it it cannot read/open ..' ?

Can you clarify?

**Comment by mckellyln** [comment]
Yes, good point.  The JIRA description states that, but this PR is just one of those - that the dafilesrv client no longer uses a certificate.  I can change the JIRA description to be more precise about what this PR is doing.

**Comment by jakesmith** [comment]
@mckellyln the commit message (and thus automatically the PR commit description and submitted) should always describe what the PR is doing too, so you don't have to go looking into the JIRA ..
Can change the commit message too.

**Comment by mckellyln** [comment]
@jakesmith will do

**Comment by jakesmith** [comment]
Thanks.
Ready to merge - @richardkchapman

**Pull request merged**